src/oscar/apps/communication/abstract_models.py

```diff
@@ -0,0 +1,213 @@
+from django.conf import settings
+from django.core.validators import RegexValidator
+from django.db import models
+from django.template import engines
+from django.template.exceptions import TemplateDoesNotExist
+from django.template.loader import get_template
+from django.utils.translation import gettext_lazy as _
+
+from oscar.apps.communication.managers import CommunicationTypeManager
+from oscar.core.compat import AUTH_USER_MODEL
+from oscar.models.fields import AutoSlugField
+
+
+class AbstractEmail(models.Model):
+    """
+    This is a record of an email sent to a customer.
+    """
+    user = models.ForeignKey(
+        AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name='emails',
+        verbose_name=_("User"),
+        null=True)
+    email = models.EmailField(_('Email Address'), null=True, blank=True)
+    subject = models.TextField(_('Subject'), max_length=255)
+    body_text = models.TextField(_("Body Text"))
+    body_html = models.TextField(_("Body HTML"), blank=True)
+    date_sent = models.DateTimeField(_("Date Sent"), auto_now_add=True)
+
+    class Meta:
+        abstract = True
+        app_label = 'communication'
```

Comment:
Is it necessary to set `app_label` and `db_table` explicitly? These look the same as the defaults that Django would produce.

Reply:
I removed `db_table`s. `app_labels` kept because of
https://github.com/django-oscar/django-oscar/blob/1b76597f8e6d11cd047c26b6235da556afd482ac/tests/integration/core/test_loading.py#L132